File: backend/hse/serializers/archive.py
from rest_framework import serializers


class ArchiveFormSerializer(serializers.Serializer):
    id = serializers.CharField()
    form_type = serializers.CharField()
    form_code = serializers.CharField()
    entry_id = serializers.IntegerField()
    form_title = serializers.CharField()
    form_number = serializers.CharField()
    project = serializers.CharField()
    created_at = serializers.DateTimeField()
    created_by = serializers.DictField(allow_null=True)
    status = serializers.CharField()
    data = serializers.DictField()

File: types/archive.ts
export interface ArchiveUserSummary {
  id: number;
  username: string;
  display_name: string;
}

export interface ArchiveListItem {
  id: string;
  entry_id: number;
  form_type: string;
  form_code: string;
  form_title: string;
  form_number: string;
  project: string | null;
  created_at: string;
  created_by: ArchiveUserSummary | null;
  status: string;
  data: Record<string, unknown>;
}

export interface ArchiveFilters {
  project?: string;
  form_type?: string;
}
